### PR TITLE
Validation dataloader so that `train_size` gets used

### PR DESCRIPTION
**Pull request by maarten-devries**
Currently, `SCVIDataModule` lacks a `val_dataloader`.

This means scVI is unable to use the `train_size: 0.9` parameter which splits the data into train and validation and allows you to log the validation loss. 

Currently, we are only able to log the train loss which is not ideal.

### CODE REVIEW

**Comment by aaronwolen** [comment]
@maarten-devries, thank you for the contribution and flagging this issue! We'll review and test whenever you're ready.

**Comment by maarten-devries** [comment]
@aaronwolen ready for review!

File: src/tiledbsoma_ml/scvi.py [hunk 1]
```diff
@@ -93,21 +105,75 @@ def __init__(
...
+                obs_column_names=self.batch_column_names,  # type: ignore[arg-type]
+                *self.dataset_args,
+                **filtered_kwargs,  # type: ignore[misc]
+            )
+            return experiment_dataloader(
+                val_dataset,
+                **self.dataloader_kwargs,
+            )
+        else:
+            return None
```
Comment (ktsitsi) [inline src/tiledbsoma_ml/scvi.py]: Given that the function is duplicate of the `train_dataloader` would it make sense to abstract the functionality under a common function `dataloader` e.g. with a selector for `train`, `val`query ids? An `enum` maybe?  In the future I would imagine a similar dataloader functionality for `test` data would also be relevant.

**Review by ktsitsi — COMMENTED** [review]
Would it be possible to fix the linter issues?

File: src/tiledbsoma_ml/scvi.py (hunk 1 above)
Comment (maarten-devries) [reply src/tiledbsoma_ml/scvi.py]: That could be nice, but keep in mind scVI does expect a function `train_dataloader` and `val_dataloader` to exist. Those could indeed call the shared helper function. It wouldn't save many lines of code though.

**Comment by maarten-devries** [comment]
@ktsitsi I did test the code above and was happy that it was working. Your suggestions around linters and shared abstractions are good but I can't address them right now.

I ended up just using `from scvi.dataloaders import TileDBDataModule` instead.

This has the validation dataloader implemented already and makes life easier. 

It probably doesn't make sense to support both `TileDBDataModule` and `SCVIDataModule` in the long term as it's potentially confusing.

**Comment by maarten-devries** [comment]
@ktsitsi 

Actually, one thing I'm noticing in favor of `SCVIDataModule` is that it's faster. So I'd prefer to use it after all!

I'm seeing a significant speedup for one epoch in a quick test (both are using `shuffle=True`):

```
GPUs   Memory   Batch Size   Data Module     Time/Epoch
-----  -------  -----------  --------------  -----------
8      80 GB    20,480       TileDBDataModule ~3:30
8      80 GB    20,480       SCVIDataModule  ~2:30
```

File: src/tiledbsoma_ml/scvi.py (hunk 1 above)
Comment (maarten-devries) [reply src/tiledbsoma_ml/scvi.py]: Addressed this now

**Comment by ktsitsi** [comment]
@maarten-devries 

Thank you so much for your changes and your contributions. Approving the PR and I am prioritizing an investigation why you noticed this performance difference between the two.

**Review by ktsitsi — APPROVED** [review]
